_drafts/floating-point-101.md

```diff
@@ -0,0 +1,441 @@
+
+This makes things a lot easier, _and_ it's true in practice. That standard's name?
+
+~~Albert Ein~~ **IEEE 754**.
+
+How's this so simple? I mean, we all know how attempts at standardization _really_ turn out.
+
+<figure>
+<a href="https://xkcd.com/927/"><img src="https://imgs.xkcd.com/comics/standards_2x.png" title="Fortunately, the charging one has been solved now that we've all standardized on mini-USB. Or is it micro-USB? Shit." alt="Standards" height="283" width="500" loading="lazy"></a>
+<figcaption><i>fig. 1</i> — There's always a relevant XKCD.</figcaption>
+</figure>
+
+You'd be correct to think more than one format must have been invented. _Plenty_ were – in the early days of computing,
+practically every system with floating-point capabilities had its own. Later on, brand-specific formats emerged: IBM
+went for hexadecimal floating-point in their mainframes, Microsoft created Microsoft Binary Format for their BASIC
+products, DEC cooked up yet something else for the VAX architecture.
+
+This changed when Intel decided in the late 70s to design the floating-point chip to rule them all – which required a
+format to rule them all too, the best possible. The decision culminated in the Intel 8087 coprocessor of 1980, but even
+before that, other companies in the space caught wind of this work and set up
+[a common effort](https://www.researchgate.net/publication/2954891_IEEE_754_An_Interview_with_William_Kahan) at the
+Institute of Electrical and Electronics Engineers (IEEE) to standardize floating-point arithmetic – the IEEE 754 working
+group. Two competing drafts prevailed: the Intel 8087 spec vs. the DEC VAX one. After some more arguments and error
+analysis, in 1981 Intel's draft won out, rapidly got adopted by everyone,
+and [the rest is history](https://www.intel.com/content/dam/www/public/us/en/documents/case-studies/floating-point-case-study.pdf)
+(though it took the committee another four years of bickering to publish that draft, of course).
+
+> For a detailed look at historic floating-point formats, see
+> [this great article by John Savard](http://www.quadibloc.com/comp/cp0201.htm).
+
+## The Specs
+
+Let's get into the meat of it, starting with… value sizes. This aspect informs the practical details of the scheme
+(range and precision), but _can't_ really be worked out from first principles. It's really just all about machine word
+sizes.
+
+Two floating-point formats are in common use today, and they both use the binary numeral system:
+
+-   32 bits, technically named `binary32`, but commonly **single precision**. Values of this size are called **floats**.
+-   64 bits, technically named `binary64`, but commonly **double precision**. Values of this size are called
+    **doubles**.
+
+Back in the day, double precision was reserved for operations requiring exceptional precision or range. With 64-bit
+processor architectures becoming standard though, it's now often the default, as its performance penalty decreased
```

Comment:
też tak językowo, może nie jest to błąd ale mi by się lepiej to zdanie czytało z "But" na początku i bez "though"